client/rest-high-level/src/test/java/org/elasticsearch/client/security/PutRoleMappingRequestTests.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security;
+
+import org.elasticsearch.client.security.support.expressiondsl.RoleMapperExpression;
+import org.elasticsearch.client.security.support.expressiondsl.fields.FieldRoleMapperExpression;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.EqualsHashCodeTestUtils;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class PutRoleMappingRequestTests extends ESTestCase {
+
+    public void testPutRoleMappingRequest() {
+        final String name = randomFrom(randomAlphaOfLength(5), null, "");
+        final boolean enabled = randomBoolean();
+        final boolean nullorEmptyRoles = randomBoolean();
+        final List<String> roles = nullorEmptyRoles ? (randomBoolean() ? null : Collections.emptyList())
+                : Collections.singletonList("superuser");
+        final RoleMapperExpression rules = randomFrom(FieldRoleMapperExpression.ofUsername("user"), null);
+        final boolean nullorEmptyMetadata = randomBoolean();
+        final Map<String, Object> metadata;
+        if (nullorEmptyMetadata) {
+            metadata = randomBoolean() ? null : Collections.emptyMap();
+        } else {
+            metadata = new HashMap<>();
+            metadata.put("k1", "v1");
+        }
+        final RefreshPolicy refreshPolicy = randomFrom(randomFrom(RefreshPolicy.values()), null);
+
+        if (Strings.hasText(name) == false) {
```

Comment:
rather than use randomization for these branches can we just test the valid case and invalid cases?